client/post-editor/controller.js

```diff
@@ -199,5 +200,30 @@ module.exports = {
 
 		page.redirect( redirectWithParams );
 		return false;
-	}
+	},
+
+	copyPost: function( context, next ) {
+		if ( ! context.query.copy ) {
+			// not copyPost, early return
+			return next();
+		}
+
+		const postId = getPostID( context );
+
+		function startEditingOnSiteSelected() {
+			const siteId = getSelectedSiteId( context.store.getState() );
+			if ( siteId ) {
+				startEditingPostCopy( context, siteId, postId ).then( () => {
+					renderEditor( context, 'post' );
+				} );
+			} else {
+				sites.once( 'change', startEditingOnSiteSelected );
+			}
+		}
+
+		startEditingOnSiteSelected();
+
+		return false;
```

Comment:
I'm currently reviewing the whole thing, as somehow today the editor doesn't show the copied content anymore. I must have accidentally changed something while wrapping up - even though I'm pretty sure I tested before pushing.
So in the review I'm gonna change everything according to your - as always - priceless annotations.

This said, I was not totally sure how `context` works, and I was a bit "shy" in adding custom data to it. Gonna experiment right now!